catalog/app/components/Preview/loaders/IFrame/LoaderBrowsable.tsx

```diff
@@ -123,71 +132,58 @@ function useSession(handle: FileHandle) {
 
   React.useEffect(() => {
     let ignore = false
-    let sessionClosure: Session | null = null
+    let sessionId: SessionId = ''
+    let timer: NodeJS.Timer
 
-    async function requestSession() {
+    async function initSession() {
       try {
-        const s = await createSession(scope, SESSION_TTL)
+        const session = await createSession(scope, SESSION_TTL)
         if (ignore) return
-        sessionClosure = s
-        setSession(s)
+        sessionId = session.id
+        setResult(AsyncResult.Ok(sessionId))
+
+        timer = setInterval(async () => {
+          try {
+            if (!sessionId) return
+            await refreshSession(sessionId, SESSION_TTL)
+          } catch (e) {
+            clearInterval(timer)
+            setResult(AsyncResult.Err(mapPreviewError(retry, e as ErrorLike)))
+            log.error(e)
+            Sentry.captureException(e)
+          }
+        }, REFRESH_INTERVAL)
       } catch (e) {
-        const err = isBucketNotBrowsableError(e as ErrorLike)
-          ? PreviewError.Forbidden()
-          : PreviewError.Unexpected({ retry })
-        setError(err)
+        clearInterval(timer)
+        setResult(AsyncResult.Err(mapPreviewError(retry, e as ErrorLike)))
+        Sentry.captureException(e)
         log.error(e)
       }
-      setLoading(false)
     }
 
-    requestSession()
+    initSession()
 
     return () => {
-      disposeSession(sessionClosure?.id)
       ignore = true
+      clearInterval(timer)
+      disposeSession(sessionId)
+      setResult(AsyncResult.Init())
```

Comment:
it's probably unsafe to call setState in a cleanup function -- react will issue a warning indicating a memory leak